### PR TITLE
feat: wallet history in new dashboard

### PR DESCRIPTION
**Pull request by nuanyang233**

### CODE REVIEW

File: packages/dashboard/src/pages/Wallets/components/History/index.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { Box, makeStyles } from '@material-ui/core'
+import { memo } from 'react'
+import { ContentContainer } from '../../../../components/ContentContainer'
+import { HistoryTable } from '../HistoryTable'
+
+const useStyles = makeStyles((theme) => ({
+    container: {
+        flex: 1,
+        padding: '0px',
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Wallets/components/History/index.tsx:9]:
```suggestion
        padding: 0,
```

File: packages/dashboard/src/pages/Wallets/components/HistoryTable/index.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+        const classes = useStyles()
+
+        return (
+            <>
+                <TableContainer className={classes.container}>
+                    {isLoading || isEmpty ? (
+                        <Box flex={1}>
+                            {isLoading ? <LoadingPlaceholder /> : null}
+                            {isEmpty ? <EmptyPlaceholder children="No Transaction history" /> : null}
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Wallets/components/HistoryTable/index.tsx:105]: `useI18N`

File: packages/dashboard/src/pages/Wallets/components/HistoryTableRow/index.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                })}
+            </TableCell>
+            <TableCell className={classes.cell} align="center">
+                <Typography fontSize="inherit">{format(transaction.timeAt, 'yyyy-MM-dd HH:mm')}</Typography>
+            </TableCell>
+            <TableCell className={classes.cell} align="center">
+                <Link
+                    href={`${resolveLinkOnExplorer(chainId)}/address/${transaction.toAddress}`}
+                    className={classes.link}>
+                    <span>{transaction.toAddress.slice(0, 5)}</span>
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Wallets/components/HistoryTableRow/index.tsx:93]: `formatEthereumAddress`

File: packages/web3-shared/src/constants/index.ts [hunk 4]
```diff
@@ -311,3 +311,50 @@ export const CONSTANTS = {
         [ChainId.Mumbai]: '0x0000000000000000000000000000000000000000',
     },
 }
+
+export const RED_PACKET_CONSTANTS = {
```
Comment (guanbinrui) [inline packages/web3-shared/src/constants/index.ts:315]: These constants should stay under the red packet plugin. Why move here?

**Comment by Jack-Works** [comment]
You need #3452 right?

**Comment by nuanyang233** [comment]
> You need #3452 right?

Yes

File: packages/dashboard/src/locales/en.json [hunk 5]
```diff
@@ -72,6 +72,11 @@
     "wallets_empty_collectible_tip": "No collectibles were found. Please add Collectibles.",
     "wallets_address_copied": "Address successfully copied",
     "wallets_address_copy": "Copy",
+    "wallets_history_types": "Types",
+    "wallets_history_value": "Value",
+    "wallets_history_time": "Time",
+    "wallets_history_receiver": "Receiver",
+    "wallets_empty_history_tips": "No Transaction history",
```
Comment (guanbinrui) [inline packages/dashboard/src/locales/en.json:79]: `No Transaction History` or `No transaction history`.

File: packages/maskbook/src/plugins/Trader/apis/trending/index.ts [hunk 6]
```diff
@@ -164,7 +164,10 @@ function isCacheExipred(dataProvider: DataProvider) {
 export async function checkAvailabilityOnDataProvider(keyword: string, type: TagType, dataProvider: DataProvider) {
     if (isBlockedKeyword(type, keyword)) return false
     // for uniswap we check availability by fetching token info dynamically
-    if (dataProvider === DataProvider.UNISWAP_INFO) await updateCache(dataProvider, keyword)
+    if (dataProvider === DataProvider.UNISWAP_INFO) {
+        console.log('done')
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/apis/trending/index.ts:168]: Remove this line.

**Review by guanbinrui — APPROVED** [review]

File: packages/dashboard/src/components/DashboardFrame/Navigation.tsx [hunk 7]
```diff
@@ -19,13 +19,14 @@ import { MaskNotSquareIcon } from '@masknet/icons'
...
 import { RoutePaths } from '../../type'
+import { omit } from 'lodash-es'
 
 const ListItemLinkUnStyled = ({ to, ...props }: ListItemProps & { to: string }) => {
     const navigate = useNavigate()
 
     return (
         <MuiListItem
-            {...props}
+            {...omit(props, 'nested')}
```
Comment (yanzhihong23) [inline packages/dashboard/src/components/DashboardFrame/Navigation.tsx:29]: need to revert this.

File: packages/dashboard/src/pages/Wallets/components/HistoryTable/index.tsx [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+                            ) : null}
+                        </Table>
+                    )}
+                </TableContainer>
+
+                {!(page === 0 && dataSource.length === 0) && !isLoading ? (
+                    <TablePagination
+                        count={-1}
+                        component="div"
+                        onPageChange={() => {}}
```
Comment (yanzhihong23) [inline packages/dashboard/src/pages/Wallets/components/HistoryTable/index.tsx:141]: no response on page change?

Comment (nuanyang233) [reply packages/dashboard/src/pages/Wallets/components/HistoryTable/index.tsx:141]: Only the back button and next button are actually used here, so the corresponding events are in `backIconButtonProps` and `nextIconButtonProps`

**Review by yanzhihong23 — APPROVED** [review]

File: packages/dashboard/src/hooks/useRedPacketAddress.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import { useConstant } from '@masknet/web3-shared'
+import { RED_PACKET_CONSTANTS } from '../constants'
+
+export function useRedPacketAddress() {
+    const addressV2 = useConstant(RED_PACKET_CONSTANTS).HAPPY_RED_PACKET_ADDRESS_V2
+    const addressV3 = useConstant(RED_PACKET_CONSTANTS).HAPPY_RED_PACKET_ADDRESS_V3
+
+    return addressV2 ?? addressV3
```
Comment (septs) [inline packages/dashboard/src/hooks/useRedPacketAddress.ts:8]:
```suggestion
    const { HAPPY_RED_PACKET_ADDRESS_V2, HAPPY_RED_PACKET_ADDRESS_V3 } = useConstant(RED_PACKET_CONSTANTS)

    return HAPPY_RED_PACKET_ADDRESS_V2 ?? HAPPY_RED_PACKET_ADDRESS_V3
```

**Comment by nuanyang233** [comment]
Covert to draft utils #3452 be merged

File: packages/dashboard/src/locales/en.json (hunk 5 above)
Comment (septs) [reply packages/dashboard/src/locales/en.json:79]: why not resolve?

File: packages/icons/general/LinkOut.tsx [hunk 10]
```diff
@@ -0,0 +1,40 @@
+import { createPaletteAwareIcon } from '../utils'
+
+export const LinkOutIcon = createPaletteAwareIcon(
+    'LinkOutIcon',
+    <>
+        <path
```
Comment (septs) [inline packages/icons/general/LinkOut.tsx:6]: plz use https://aka.pw/svgomg reduce svg size

File: packages/maskbook/src/plugins/Trader/apis/trending/index.ts (hunk 6 above)
Comment (septs) [reply packages/maskbook/src/plugins/Trader/apis/trending/index.ts:168]: @nuanyang233

File: packages/icons/general/LinkOut.tsx (hunk 10 above)
Comment (nuanyang233) [reply packages/icons/general/LinkOut.tsx:6]: It has been reduced

File: packages/dashboard/src/locales/en.json (hunk 5 above)
Comment (septs) [reply packages/dashboard/src/locales/en.json]: @nuanyang233 you unsolved this problem 
@guanbinrui need change to `No Transaction History` or `No transaction history` not is `No transaction history.`

Comment (nuanyang233) [reply packages/dashboard/src/locales/en.json]: Sorry for my carelessness.

File: packages/dashboard/stories/components/Wallet/HistoryTable.tsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
+import { story } from '@masknet/storybook-shared'
+import { HistoryTableUI as C } from '../../../src/pages/Wallets/components/HistoryTable'
+import { action } from '@storybook/addon-actions'
+import { ZerionTransactionDirection } from '@masknet/web3-shared'
+
+const { meta, of } = story(C)
+
+export default meta({ title: 'Components/Wallet/History Table' })
```
Comment (Jack-Works) [inline packages/dashboard/stories/components/Wallet/HistoryTable.tsx:8]: This story fails "Component must be wrapped with <Container.Provider>"

**Comment by Jack-Works** [comment]
mark this as not ready for review.

File: packages/dashboard/src/pages/Wallets/components/HistoryTableRow/index.tsx [hunk 12]
```diff
@@ -0,0 +1,100 @@
+import { memo } from 'react'
+import format from 'date-fns/format'
```
Comment (septs) [inline packages/dashboard/src/pages/Wallets/components/HistoryTableRow/index.tsx:2]:
```suggestion
import formatDateTime from 'date-fns/format'
```

File: packages/dashboard/src/pages/Wallets/components/HistoryTable/index.tsx (hunk 8 above)
Comment (septs) [reply packages/dashboard/src/pages/Wallets/components/HistoryTable/index.tsx:141]: plz use `lodash.noop`

File: packages/dashboard/src/pages/Wallets/components/TokenTable/index.tsx [hunk 13]
```diff
@@ -102,7 +103,7 @@ export const TokenTableUI = memo<TokenTableUIProps>(
                             {isEmpty ? <EmptyPlaceholder children={t.wallets_empty_tokens_tip()} /> : null}
                         </Box>
                     ) : (
-                        <Table stickyHeader sx={{ padding: '0 44px' }}>
+                        <Table stickyHeader sx={{ padding: '0px 44px' }}>
```
Comment (septs) [inline packages/dashboard/src/pages/Wallets/components/TokenTable/index.tsx:106]: plz revert the changes.

**Review by septs — CHANGES_REQUESTED** [review]

**Review by septs — DISMISSED** [review]

File: packages/dashboard/src/hooks/useRedPacketAddress.ts [hunk 14]
```diff
@@ -0,0 +1,7 @@
+import { useRedPacketConstants } from '@masknet/web3-shared'
+
+export function useRedPacketAddress() {
+    const { HAPPY_RED_PACKET_ADDRESS_V2, HAPPY_RED_PACKET_ADDRESS_V3 } = useRedPacketConstants()
```
Comment (guanbinrui) [inline packages/dashboard/src/hooks/useRedPacketAddress.ts:4]: @zhouhanseng Please review this.

File: packages/dashboard/src/pages/Wallets/components/HistoryTable/index.tsx [hunk 15]
```diff
@@ -0,0 +1,162 @@
...
+
+export const HistoryTable = memo(() => {
+    const [page, setPage] = useState(0)
+    const chainId = useChainId()
+    const account = useAccount()
+    const {
+        value = { transactions: [], hasNextPage: false },
+        loading: transactionLoading,
+        error: transactionError,
+    } = useTransactions('0x0d09dc9a840b1b4ea25194998fd90bb50fc2008a', page, 50)
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Wallets/components/HistoryTable/index.tsx:65]: A dead code address?

File: packages/dashboard/src/hooks/useRedPacketAddress.ts (hunk 14 above)
Comment (zhouhanseng) [reply packages/dashboard/src/hooks/useRedPacketAddress.ts:4]: okay

**Review by guanbinrui — DISMISSED** [review]

File: packages/dashboard/src/pages/Wallets/components/TransactionIcon/index.tsx [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+
+export interface TransactionIconProps {
+    transactionType: string
+    address: string
+    failed: boolean
+}
+
+export const TransactionIcon = memo<TransactionIconProps>(({ address, failed, transactionType }) => {
+    const redPacketAddress = useRedPacketAddress()
+    const isRedPacket = isSameAddress(redPacketAddress, address)
```
Comment (zhouhanseng) [inline packages/dashboard/src/pages/Wallets/components/TransactionIcon/index.tsx:40]: I think [web3.eth.getTransactionReceipt(hash [, callback])](https://web3js.readthedocs.io/en/v3.0.0-rc.5/web3-eth.html#gettransactionreceipt) is a better way to determine is this tx a redpacket or ITO or something else. The contract address on different network may be the same ([How is the address of an Ethereum contract computed?](https://ethereum.stackexchange.com/questions/760/how-is-the-address-of-an-ethereum-contract-computed)).  
But the current solution is more simple and acceptable since there's no same red-packet contract addresses in our codebase, and we can and should ensure it in the future.

**Review by zhouhanseng — APPROVED** [review]

File: packages/dashboard/src/hooks/useRedPacketAddress.ts (hunk 14 above)
Comment (zhouhanseng) [reply packages/dashboard/src/hooks/useRedPacketAddress.ts:4]: How about a transaction of `HAPPY_RED_PACKET_ADDRESS_V1`？I think this hook is unnecessary, you can use `useRedPacketConstants()` directly, then write a function to compare the address with V1 V2 and V3

**Review by zhouhanseng — CHANGES_REQUESTED** [review]

**Comment by nuanyang233** [comment]
I will revisit the red packet address. The reason for converting to draft is multi-chain tx history is not currently supported. CC @guanbinrui @zhouhanseng

File: packages/dashboard/stories/components/Wallet/HistoryTable.tsx (hunk 11 above)
Comment (Jack-Works) [reply packages/dashboard/stories/components/Wallet/HistoryTable.tsx:7]: Is this fixed?

Comment (nuanyang233) [reply packages/dashboard/stories/components/Wallet/HistoryTable.tsx:7]: yes

**Review by Jack-Works — APPROVED** [review]

File: packages/dashboard/src/pages/Wallets/components/TransactionIcon/index.tsx [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+export const TransactionIcon = memo<TransactionIconProps>(({ address, failed, transactionType }) => {
+    const { HAPPY_RED_PACKET_ADDRESS_V1, HAPPY_RED_PACKET_ADDRESS_V2, HAPPY_RED_PACKET_ADDRESS_V3 } =
+        useRedPacketConstants()
+
+    const isRedPacket =
+        isSameAddress(HAPPY_RED_PACKET_ADDRESS_V1, address) ||
+        isSameAddress(HAPPY_RED_PACKET_ADDRESS_V2, address) ||
+        isSameAddress(HAPPY_RED_PACKET_ADDRESS_V3, address)
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Wallets/components/TransactionIcon/index.tsx:44]: It's stabled because of #3809. But we can update it later.

**Review by guanbinrui — APPROVED** [review]
